libpod/pod.go

```diff
@@ -138,46 +103,109 @@ func (p *Pod) Namespace() string {
 
 // ResourceLim returns the cpuset resource limits for the pod
 func (p *Pod) ResourceLim() *specs.LinuxResources {
-	resCopy := &specs.LinuxResources{}
-	if err := JSONDeepCopy(p.config.InfraContainer.ResourceLimits, resCopy); err != nil {
-		return nil
-	}
-	if resCopy != nil && resCopy.CPU != nil {
-		return resCopy
-	}
+	resCopy := &specs.Linux{}
 	empty := &specs.LinuxResources{
 		CPU: &specs.LinuxCPU{},
 	}
+	infra, err := p.runtime.GetContainer(p.state.InfraContainerID)
+	if err != nil {
+		return empty
+	}
+	conf, err := infra.specFromState()
+	if err != nil {
+		return empty
+	}
+	if err = JSONDeepCopy(conf.Linux, resCopy); err != nil {
+		return nil
+	}
+	if resCopy != nil && resCopy.Resources != nil && resCopy.Resources.CPU != nil {
+		return resCopy.Resources
+	}
+
 	return empty
 }
 
 // CPUPeriod returns the pod CPU period
 func (p *Pod) CPUPeriod() uint64 {
-	resCopy := &specs.LinuxResources{}
-	if err := JSONDeepCopy(p.config.InfraContainer.ResourceLimits, resCopy); err != nil {
+	resCopy := &specs.Spec{}
+	//resCopy.Linux = &specs.Linux{}
+	//resCopy.Linux.Resources = &specs.LinuxResources{}
+	if p.state.InfraContainerID == "" {
+		return 0
+	}
+	infra, err := p.runtime.GetContainer(p.state.InfraContainerID)
+	if err != nil {
+		return 0
+	}
+	conf := infra.Config()
+	if err := JSONDeepCopy(conf.Spec, resCopy); err != nil {
```

Comment:
No need - we deep-copy structs that we return, but you're not returning a struct, just a uint64